cardano-cli/src/Cardano/CLI/EraBased/Transaction/Option.hs

```diff
@@ -188,7 +189,7 @@ pTransactionBuildCmd sbe envCli = do
         <*> optional pTotalCollateral
         <*> many pTxOut
         <*> pChangeAddress
-        <*> optional (pMintMultiAsset sbe AutoBalance)
+        <*> (fmap join . optional $ pMintMultiAsset sbe AutoBalance)
```

Comment:
You can parameterize `pMintMultiAsset` on `MaryEraOnwards era`, remove the `Maybe` from the type  signature and call `caseShelleyToAllegraOrMaryEraOnwards` instead. I think that is cleaner but I wouldn't block the PR on it.